### DISCUSSION TITLE
Warn on git version < 2.22 (invalid line x: reset)

### DISCUSSION THREAD

**Discussion by annasun28** [arxanas/git-branchless#368] [Ideas]
It seems that git move (and git restack) don't work prior to git version 2.22, since they insert `reset` into the interactive rebase todo list. I know the wiki says that versions prior to 2.24 are untested, but it would be nice to have a warning (I ran into this because I forgot to check the git version on the third machine I installed git-branchless onto, and had quite a hard time figuring out why `reset` was not a valid command in the todo list - this is difficult to google for since most search results containing "git"  and "reset" are unrelated to interactive rebase or --rebase-merges)

**Comment by annasun28** [comment]
<img width="606" alt="image" src="https://user-images.githubusercontent.com/13106449/164583856-a4278987-8ff1-4278-8569-09f497b5d066.png">
Some ineffective searches

**Comment by arxanas** [comment]
Hmm, it's a bit strange, because it seems like `reset` existed as early as 2.18: https://git-scm.com/docs/git-rebase/2.18.0

It's possible that we instantiated the on-disk way such that 2.22 can't process it but later versions can. (I recall that some earlier versions arbitrarily didn't work if you didn't set the `--interactive` flag, even though the rebase plan was the same.)

When you run `git branchless init`, it should check the Git version and produce a warning below version 2.29 that `git undo` won't work. Do you recall seeing that? We could add another version check there.
